common/djangoapps/philu_commands/management/commands/tests/test_auto_generate_certificates_for_open_courses.py

```diff
@@ -0,0 +1,133 @@
+
+        mock_log_info.side_effect = assert_certificates_generated
+        call_command('auto_generate_certificates_for_open_courses')
+        mock_log_info.assert_called(2)
+
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.modulestore')
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.log.info')
+    def test_no_open_course_available(self, mock_log_info, mock_modulestore):
+        """
+        Test 'If no open course is available'
+        """
+        mock_response = Mock(**{"get_courses.return_value": [self.course_1]})
+        mock_modulestore.return_value = mock_response
+        call_command('auto_generate_certificates_for_open_courses')
+        assert not mock_log_info.called
+
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.modulestore')
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.log.info')
+    def test_course_not_valid_for_certificate_auto_generation(self, mock_log_info, mock_modulestore):
+        """
+        Test 'If course is not valid'
+        """
+        mock_response = Mock(**{"get_courses.return_value": [self.course_1]})
+        mock_modulestore.return_value = mock_response
+        call_command('auto_generate_certificates_for_open_courses')
+        assert not mock_log_info.called
+
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.modulestore')
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.log.info')
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.'
+           'is_course_valid_for_certificate_auto_generation')
+    def test_course_valid_cert_data_not_exists(self, mock_is_course_valid, mock_log_info,
+                                               mock_modulestore):
+        """
+        Test 'If course is valid but certification data does not exist'
+        """
+        mock_response = Mock(**{"get_courses.return_value": [self.course_1]})
+        mock_modulestore.return_value = mock_response
+        mock_is_course_valid.return_value = True
+        call_command('auto_generate_certificates_for_open_courses')
+        assert mock_log_info.called
+
+    @patch('philu_commands.management.commands.auto_generate_certificates_for_open_courses.has_active_certificate',
+           return_value=True)
+    def test_is_course_valid_for_certificate_auto_generation(self, mock_has_active_certificate):
+        """
+        Test 'check if course is in active state'
+        """
+        self.course_1.has_started = Mock(return_value=True)
+        self.course_1.has_ended = Mock(return_value=False)
+        self.course_1.may_certify = Mock(return_value=True)
+        mock_has_active_certificate.return_value = True
+        assert is_course_valid_for_certificate_auto_generation(self.course_1) is True
```

Comment:
I think **"is True"** is redundant. `assert is_course_valid_for_certificate_auto_generation(self.course_1)` will work